add method to File to get extents on-disk

agrif opened the pull request:
Hello!

I've added a method to `File` to iterate over the extents of the file data, a la [bmap](https://manpages.debian.org/testing/linux-manual-4.9/bmap.9). I opted to expose the extents as bytes, just because that seemed the simplest.

I'm playing around with bootloaders, and I use this information to load a binary off a FAT32 disk from a stub that does not have enough code space to include a real FAT32 implementation.

Let me know if I can improve this PR!

rafalh commented on src/file.rs:
```diff
@@ -25,6 +25,15 @@ pub struct File<'a, IO: ReadWriteSeek, TP, OCC> {
 
+/// An extent containing a file's data on disk.
+///
+/// This is created by the `extents` method on `File`. All values in bytes.
```
Missing verb: All values ARE in bytes.
Perhaps more detailed description what you mean by an extent would be a good idea like in `extents` fun doc (the regions on-disk occupied by this file.). Perhaps changing it to "byte ranges" would be better? Regions reminds me of winapi regions which are something very different

rafalh commented:
To be honest I feel a little uneasy about this change because it is useful only in a very specific use-case that I think is not popular  and it feels a bit low level compared to the rest of library but I am willing to give it a try.

agrif commented:
I've updated the docs on `Extent` and switched to using 'byte ranges' instead of regions.

If you think this is outside the scope of this library, that's ok! I just needed this info myself, and figured I'd open a PR with the small change I made to expose it. No worries!

It's certainly low-level, but just to show I'm not completely on the wrong track, this is also the method GRUB uses to load stage2 off disk from the tiny MBR stage1.

rafalh commented:
I forgot to merge it :D Sorry for the delay and thanks for your contribution.
